github/data_source_github_repository_branches.go

```diff
@@ -0,0 +1,70 @@
+package github
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v43/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceGithubRepositoryBranches() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGithubRepositoryBranchesRead,
+		Schema: map[string]*schema.Schema{
+			"repository": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"branches": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"protected": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func flattenBranches(branches []*github.Branch) []interface{} {
+	if branches == nil {
+		return []interface{}{}
+	}
+
+	branchList := make([]interface{}, 0, len(branches))
+	for _, branch := range branches {
+		branchMap := make(map[string]interface{})
+		branchMap["name"] = branch.GetName()
+		branchMap["protected"] = branch.GetProtected()
+		branchList = append(branchList, branchMap)
+	}
+
+	return branchList
+}
+
+func dataSourceGithubRepositoryBranchesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Owner).v3client
+	orgName := meta.(*Owner).name
+	repoName := d.Get("repository").(string)
+
+	branches, _, err := client.Repositories.ListBranches(context.TODO(), orgName, repoName, nil)
```

Comment:
note, this data source will only display the first 30 branches of a repository, this is the default setting of the [Github API endpoint](https://docs.github.com/en/rest/reference/branches#list-branches) 
<img width="1228" alt="Screen Shot 2022-04-18 at 4 25 30 PM" src="https://user-images.githubusercontent.com/15153943/163892391-b3cecb4a-ed2a-4e66-bd39-1f8643bd6dc8.png">

This mimics the previous behavior introduced in https://github.com/integrations/terraform-provider-github/pull/892: https://github.com/kzhaotest/branches_test1 (see in this repo although there are 50+ branches, only 30+ are shown in the README (tf located [here](https://github.com/kzhaotest/branches/tree/main/before))).

We can introduce pagination in a future PR, but for now just replicate the previous behavior